common/src/main/java/com/wynntils/screens/settings/WynntilsBookSettingsScreen.java

```diff
@@ -349,19 +350,19 @@ private void reloadConfigurableButtons() {
 
         Category oldCategory = null;
 
-        List<Configurable> configurableList = Stream.concat(
-                        Managers.Feature.getFeatures().stream()
-                                .filter(feature -> searchMatches(feature)
-                                        || feature.getVisibleConfigOptions().stream()
-                                                .anyMatch(this::configOptionContains))
-                                .map(feature -> (Configurable) feature)
-                                .sorted(),
-                        Managers.Overlay.getOverlays().stream()
-                                .filter(overlay -> searchMatches(overlay)
-                                        || overlay.getVisibleConfigOptions().stream()
-                                                .anyMatch(this::configOptionContains))
-                                .sorted())
-                .toList();
+        List<Configurable> configurableList = Managers.Feature.getFeatures().stream()
+                .filter(feature -> searchMatches(feature)
+                        || feature.getVisibleConfigOptions().stream().anyMatch(this::configOptionContains))
+                .map(feature -> (Configurable) feature)
+                .sorted()
+                .collect(Collectors.toList());
+
+        configurableList.addAll(Managers.Overlay.getOverlays().stream()
+                .filter(overlay -> !configurableList.contains(Managers.Overlay.getOverlayParent(overlay)))
+                .filter(overlay -> searchMatches(overlay)
+                        || overlay.getVisibleConfigOptions().stream().anyMatch(this::configOptionContains))
+                .sorted()
```

Comment:
Just out of intellectual curiosity, I wonder if this could have been done purely with streams? If you had concatenated the streams and then sorted them?

Reply:
I don't think so, since I needed the features to be collected first, so I can check for not adding duplicates (as adding features will add all of its overlays, but adding an overlay will not add the holder feature..)

Reply:
(Which might not be the smartest decision to do, but we are going through iterations of this :) and it is getting better..)